keylime/tenant.py

```
@@ -1114,8 +1109,17 @@ def do_cvdelete(self, verifier_check: bool = True) -> None:
-                # Marked for deletion (need to modify the code on CI tests)
                 logger.info("Agent %s deleted from the CV", self.agent_uuid)
+        elif response_json["code"] == 404:
+            # The agent was already deleted (e.g. by another caller
+            # between the do_cvstatus check and the DELETE request).
+            # The desired end state is reached — treat as success.
+            logger.info("Agent %s is already absent from the CV", self.agent_uuid)
+        else:
+            raise UserError(
+                f"Unexpected response code {response_json['code']} from "
+                f"{self.verifier_fid_str} while deleting {self.agent_fid_str}"
+            )
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)